CorsixTH/Src/th_lua_iso.cpp

```diff
@@ -80,11 +80,27 @@ int l_isofs_file_size(lua_State* L) {
   return 1;
 }
 
+int l_isofs_file_offsets(lua_State* L) {
+  iso_filesystem* pSelf = luaT_testuserdata<iso_filesystem>(L);
+  const char* sFilename = luaL_checkstring(L, 2);
+  iso_filesystem::file_handle iFile = pSelf->find_file(sFilename);
+  if (!iso_filesystem::is_handle_good(iFile)) {
+    lua_pushnil(L);
+    lua_pushnil(L);
+    lua_pushfstring(L, "Could not find \'%s\' in .iso image", sFilename);
+    return 3;
+  }
+  uint32_t start = pSelf->get_file_start(iFile);
+  lua_pushinteger(L, start);
+  lua_pushinteger(L, start + pSelf->get_file_size(iFile));
```

Comment:
It isn't documented, but `start` looks like an inclusive lower bound. (If `.sector` is 0 you get 0 as return value).
Assuming "size" is a proper size, the second value is thus an exclusive upper bound.

In Lua it is normal to use inclusive upper bounds. (and start at 1, but I can see that might be a bad idea).

As it's used for building a string, it's probably fine just a bit strange from a Lua point of view.

Reply:
I'd rather cover this one with documentation instead of converting back and forth.

Reply:
@Alberth289346 for what it's worth, lua doesn't extend this convention to file offsets. https://www.lua.org/manual/5.4/manual.html#pdf-file:seek

Reply:
Interesting :)